strawberry/cli/commands/schema_importer/sdl_transpiler.py

```diff
@@ -0,0 +1,210 @@
+    {%- for field in ast.fields %}
+    {{ get_field_attribute(field) }}
+    {%- endfor %}
+    {%- endif -%}
+    {%- if ast.kind == 'enum_type_definition' -%}
+    {%- for enum in ast.values %}
+    {{ enum.name.value }} = '{{ enum.name.value.lower() }}'
+    {%- endfor %}
+    {%- endif -%}
+"""
+
+# strawberry union definitions
+UNION_TEMPLATE = """{{ class_name }} = {{ get_union(ast) }}"""
+
+# strawberry directives
+# TODO: How do i know which type should first arg(value) be ?
+DIRECTIVE_TEMPLATE = dedent(
+    """\
+@strawberry.directive(
+    locations=[
+        {%- for field in ast.locations %}
+        DirectiveLocation.{{ field.value }}
+        {%- endfor %}
+    ],
+{% if description -%}{{ '    ' + description[1:-1] + '\n' }}{%- endif -%}
+)
+def {{ class_name }}(
+    {%- for field in ast.arguments %}
+    {{ get_field_attribute(field) }}
+    {%- endfor %}
+):
+    pass
+"""
+)
+
+# QUESTION: Is there a better way to determine this?
+SCALAR_TYPES = {
+    "Int": "int",
+    "String": "str",
+    "Float": "float",
+    "Boolean": "bool",
+    "ID": "strawberry.ID",
+}
+
+# Base decorator kinds
+DECORATOR_KINDS = {
+    "union_type_definition": "",
+    "schema_definition": "@strawberry.type",
+    "enum_type_definition": "@strawberry.enum",
+    "object_type_definition": "@strawberry.type",
+    "directive_definition": "@strawberry.directive",
+    "input_object_type_definition": "@strawberry.input",
+    "interface_type_definition": "@strawberry.interface",
+    # What about a mutation?
+}
+
+
+def get_class_name(ast):
+    name = "Schema" if ast.kind == "schema_definition" else ast.name.value
+    name += "(Enum)" if ast.kind == "enum_type_definition" else ""
+    return name
+
+
+def get_decorator(ast):
+    """Creates and returns decorator string"""
+    return DECORATOR_KINDS[ast.kind]
+
+
+def get_description(ast):
+    """Creates and returns decorator string"""
+    if not hasattr(ast, "description") or ast.description is None:
+        return ""
+    else:
+        return f"(description='''{ast.description.value}''')"
+
+
+def get_union(ast):
+    """Format union type"""
+    types = "(" + ", ".join((t.name.value for t in ast.types)) + ")"
+    description = get_description(ast)
+    description = f"{description[1:-1]}" if description else ""
+    union_type = "strawberry.union({}{}{})".format(
+        f"\n    '{get_class_name(ast)}',",
+        f"\n    {types},",
+        f"\n    {description}\n" if description else "\n",
+    )
+
+    return union_type
+
+
+def get_field_attribute(field: Union[FieldDefinitionNode, InputValueDefinitionNode]):
+    """
+    Format and return a whole attribute string
+    consists of attribute name in snake case and field type
+    """
+    field_name = get_field_name(field.name.value)
+    field_type = get_field_type(field)
+    is_custom_type = not any(
+        builtin_type in field_type
+        for builtin_type in ("str", "int", "float", "ID", "bool")
+    )
```

Comment:
Am I correct in understanding that `field_type` is a string? Any custom type with any of these as a _substring_ (e.g. if I had a type named "winter" or something), would make `is_custom_type == False` no?

Reply:
You're correct. Given a string like `List[int]` it tries to detect if it uses a custom type with this logic. But it does have a problem you just found. Can you think of a better solution?

Reply:
Sorry for the delayed response here. Could you do something like `field_type == "int" or "[int]" in field_type` (obviously expanded for all the types here)?